File: spec/python/test_process_rotate.py
import unittest

from process_rotate import ProcessRotate

class TestProcessRotate(unittest.TestCase):
    def test_process_rotate(self):
        with ProcessRotate.from_file("src/process_rotate.bin") as r:

            self.assertEqual(r.buf1, b"Hello")
            self.assertEqual(r.buf2, b"World")
            self.assertEqual(r.buf3, b"There")

File: spec/lua/test_process_rotate.lua
local luaunit = require("luaunit")

require("process_rotate")

TestProcessRotate = {}

function TestProcessRotate:test_process_rotate()
    local r = ProcessRotate:from_file("src/process_rotate.bin")

    luaunit.assertEquals(r.buf1, "Hello")
    luaunit.assertEquals(r.buf2, "World")
    luaunit.assertEquals(r.buf3, "There")
end
